qiskit/assembler/assemble_schedules.py

```diff
@@ -246,20 +246,29 @@ def _validate_meas_map(instruction_map: Dict[Tuple[int, instructions.Acquire],
     Raises:
         QiskitError: If the instructions do not satisfy the measurement map.
     """
+    sorted_inst_map = sorted(instruction_map.items(), key=lambda item: item[0])
     meas_map_sets = [set(m) for m in meas_map]
-
-    # Check each acquisition time individually
-    for _, instrs in instruction_map.items():
-        measured_qubits = set()
-        for inst in instrs:
-            measured_qubits.add(inst.channel.index)
-
-        for meas_set in meas_map_sets:
-            intersection = measured_qubits.intersection(meas_set)
-            if intersection and intersection != meas_set:
-                raise QiskitError('Qubits to be acquired: {} do not satisfy required qubits '
-                                  'in measurement map: {}'.format(measured_qubits, meas_set))
-
+    all_measured_qubits = [inst.channel.index for _, insts in sorted_inst_map
+                           for inst in insts]
+
+    # 1. if the qubits are not in the meas_map -- Raise Error
+    if not set(all_measured_qubits).issubset(set.union(*meas_map_sets)):
+        raise QiskitError("The measured qubits - {} is not in the measurement "
+                          "map - {}".format(all_measured_qubits, meas_map))
+    # 2. if there is time overlap:
+    #    - if the overlap is in the same meas_map -- Raise Error
+    for idx, instr in enumerate(sorted_inst_map[:-1]):
+        inst_end_time = instr[0][0] + instr[0][1]
+        if sorted_inst_map[idx+1][0][0] < inst_end_time:
```

Comment:
```suggestion
        next_instr = sorted_inst_map[idx+1]
        next_instr_time = next_instr[0][0]
        if next_instr_time < inst_end_time:
```
We got a bit inconsistent about abbreviating as `inst` or `instr`. Either is ok -- let's just go with `inst` everywhere?

Reply:
Fixed in https://github.com/Qiskit/qiskit-terra/pull/5971/commits/e471d666e83904922779f399389fee80600b0fca and https://github.com/Qiskit/qiskit-terra/pull/5971/commits/22999a40daf444428af9fbd6f7c66b5eef75c5ce